examples/bri-3/src/bri/merkleTree/agents/merkleTree.agent.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable } from '@nestjs/common';
+import MerkleTree from 'merkletreejs';
+import { v4 } from 'uuid';
+import { BpiMerkleTree } from '../models/bpiMerkleTree';
+import { MerkleTreeStorageAgent } from './merkleTreeStorage.agent';
+import * as crypto from 'crypto';
+
+@Injectable()
+export class MerkleTreeAgent {
+  constructor(private storageAgent: MerkleTreeStorageAgent) {}
+  public createNewMerkleTree(
+    leaves: string[],
+    hashFunction: string,
+  ): BpiMerkleTree {
+    const hashHelper = (data: any): Buffer => {
+      return crypto.createHash(hashFunction).update(data).digest();
+    };
+
+    const hashedLeaves = leaves.map((x) => hashHelper(x));
+
+    return new BpiMerkleTree(v4(), new MerkleTree(hashedLeaves, hashHelper));
```

Comment:
can hashHelper just be part of MerkleTree class since it is used there in both creation and update? that way you pass leaves to merkle tree and hashing happens inside the class

again, up to you, if there is a need to provide hashHelper and hashedLeaves to merkle tree then it should be fine

Reply:
It can be done that way - it does complicate the mapping done when the object is returned from the DB. The merkleTree comes from the DB as  string and needs to be unmarshaled into the MerkleTree - the automapper would be using the default constructor at that point and would expect a string array of leaves. It can be done of course with something like introducing another DTO/constructor, a more complicated map, etc. I will move it into the domain object in successive PRs once merkletree module is more built and leave as a TODO for now.